### DISCUSSION TITLE
[VSCode] missing context menu and retry option, as a result no connection behind company proxy

### DISCUSSION THREAD

**Discussion by FrostKiwi** [community/community#35263] [Copilot Conversations]
In Visual Studio Code

There used to be a context menu when clicking the copilot extension's bottom logo to attempt a connection retry.
![image](https://user-images.githubusercontent.com/60887273/193953581-926e61a9-002d-4041-ae12-60f2cbec3747.png)
This context menu does not appear anymore. I'm behind a company proxy and when behind said proxy, some programs take two tries to connect, as in the first one always fails. Copilot is one of them. Initially when launching VSCode, copilot is unable to connect, but when clicking retry, it successfully connects. It's a bizzare system wide effect.

However, Copilot's context retry button does not seem to exist anymore and going through all the keybinds that copilot exposes, there also seems to be no retry command.

How can I perform this retry, that I used to be able to do?

Does this issue occur when all extensions are disabled?: Yes (With only Copilot extension running)

<details><summary>Version Details</summary>

```
Version: 1.71.2 (user setup)
Commit: 74b1f979648cc44d385a2286793c226e611f59e7
Date: 2022-09-14T21:03:37.738Z
Electron: 19.0.12
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.19044
Sandboxed: No
```
</details>

Also: Where is the repo for the extension? There is this community tab, but is it actually the right place to post extension related issues, because the issue in the VSCode issue https://github.com/microsoft/vscode/issues/162706 got closed. (As I expected after not finding a proper place to post issues)

**Comment by iamcjwilliams** [comment]
@FrostKiwi Our copilot team has been notified

**Reply by FrostKiwi** [reply]
@iamcjwilliams 
Thank you <3
Wanted to add, that Copilot actually fails silently under a certain situation. No error, Copilot Logo having the connected-icon-status, but failing to do anything.
![image](https://user-images.githubusercontent.com/60887273/196307305-da49df30-63ee-4780-8905-f2b8ba657eb5.png)

(Windows 10 btw) Here is how Copilot worked for me before the context menu was removed:
There is a company defined system wide proxy, which some programs pick up on. Copilot being one of them. Usually goes like: start vsc-> connection fail -> retry -> Internet access.
Some don't ( https://github.com/ankitects/anki for instance) and I provide the HTTP_PROXY environment variable to help them out, by copy pasting the same proxy info into that variable. Works great.
But if this variable is defined, Copilot fails silently. Copilot starts and just has the connected logo, but nothing happens. No code suggestions, no error output.

So what I used to do is the following: 
 - Change the ENV variable HTTP_PROXY to a different name
 - start VSC with Copilot,which connects successfully after pressing retry
 - change HTTP_PROXY back to it's original "HTTP_PROXY" name , so that the rest of my programs which don't properly read the system proxy can connect as well.

But as already mentioned, with the retry button removed, this doesn't work anymore. Using a company external proxy-less internet connection work fine though.
![image](https://user-images.githubusercontent.com/60887273/196308575-50afd1dc-2639-4ba6-9e01-6a719c2af9cd.png)

**Comment by ossanna16** [comment]
Thank you for asking a great question @FrostKiwi! I'm glad you found our community 🚀

**Comment by kimadeline** [comment]
Hi @FrostKiwi, going to address a couple of points there:

- The notification with a "Retry" button is still there, but only appears if the extension failed to activate somehow, with a message along the lines of `GitHub Copilot could not connect to server. Extension activation failed.` Since you don't get the message, it means that the extension activated successfully. If you still want to mimic a retry experience, the best way for now would be to disable and then re-enable the extension, that would force it to deactivate and reactivate itself.

- Proxy: Do you specify proxy info only via the `HTTP_PROXY` environment variable, or do you also use the VS Code proxy settings?
![image](https://user-images.githubusercontent.com/51720070/197598757-e79e64ee-54f6-4d00-909e-3e43beffb35c.png)
Could you open the dev tools window (Command Palette > `Developer: Toggle Developer Tools`) and see if there are any Copilot-related messages in there?

- Is this the best place to post Copilot-related questions? For now, yes 🙂 We don't have a public repo for folks to open issues at the moment.

**Reply by FrostKiwi** [reply]
@kimadeline 
> Since you don't get the message, it means that the extension activated successfully.

Thx for clearing things up. So the real bug happening is thus [here](https://github.com/orgs/community/discussions/35263#discussioncomment-3901112):
> But if this variable is defined, Copilot fails silently. Copilot starts and just has the connected logo, but nothing happens. No code suggestions, no error output.

> that would force it to deactivate and reactivate itself.

I actually tried that as part of my troubleshooting, before posting this, but it doesn't act the same as a retry under my specific circumstance behind this specific proxy. This is something peculiar to the way the company VPN, Proxy is setup or some weird interaction with Windows or it's security system. The first time connecting, some programs time-out on their *first* connection attempt. A retry within the program results in a successful connection, but restarting said program does not. The same seems to be happening with retrying from within Copilot vs force-reloading the whole extension. With force reloading: no error in Copilot and it's just stuck doing nothing. Which I guess is the actual bug on display here: the silent failing.

> Do you specify proxy info only via the HTTP_PROXY environment variable, or do you also use the VS Code proxy settings?

Didn't know about the HTTP_PROXY variable within VSCode until now. Never tried that. Unfortunately my trial ran out 3 days ago^^

**Reply by kimadeline** [reply]
Oh, that's unfortunate. If you pick Copilot back up eventually, feel free to ping me again, and we can try to work through your proxy config 🙂
